AllJoyn/Platform/Bridge/BridgeCore/AllJoynHelper.cpp

```diff
@@ -368,10 +353,42 @@ QStatus AllJoynHelper::SetMsgArg(_In_ std::shared_ptr<IAdapterValue> adapterValu
     return status;
 }
 
-QStatus AllJoynHelper::SetMsgArg(_Inout_ alljoyn_msgarg msgArg, _In_ const std::string& ajSignature, _In_ std::vector<bool>& arrayArg)
+template<typename T>
+QStatus AllJoynHelper::SetMsgArg(
+    _Inout_ alljoyn_msgarg msgArg, _In_ const std::string& ajSignature, _In_ std::vector<T>& arrayArg)
+{
+    QStatus status = ER_OK;
+
+    if (ajSignature.length() != 2 || ajSignature[0] != 'a')
+    {
+        status = ER_BAD_ARG_2;
+        goto leave;
+    }
+
+    if (!arrayArg.empty())
+    {
+        status = alljoyn_msgarg_set_and_stabilize(msgArg, ajSignature.c_str(), arrayArg.size(), arrayArg.data());
+    }
+    else
+    {
+        status = alljoyn_msgarg_set(msgArg, ajSignature.c_str(), 0, nullptr);
```

Comment:
The correct value should be a 0-length array, not a 1-length array with a default value.
(The code in this file that dealt with arrays was apparently never tested, since it was buggy. I'm fixing the bugs.)

---
In reply to: [67949892](https://github.com/tjaffri/msiot-samples/pull/7#discussion_r67949892) [](ancestors = 67949892)